examples/templates/do-linux/main.tf

```diff
@@ -11,77 +11,205 @@ terraform {
-  type        = number
+data "coder_parameter" "step2_do_admin_ssh_key" {
+  name        = "Enter admin SSH key ID (some Droplet images require an SSH key to be set):"
   description = <<-EOF
-    Enter admin SSH key ID (some Droplet images require an SSH key to be set):
-
     Can be set to "0" for no key.
 
     Note: Setting this to zero will break Fedora images and notify root passwords via email.
 
       $ doctl compute ssh-key list
   EOF
-  sensitive   = true
-
+  type        = "number"
+  default     = "0"
+  mutable     = false
   validation {
-    condition     = var.step2_do_admin_ssh_key >= 0
-    error_message = "Invalid Digital Ocean SSH key ID, a number is required."
+    min = 0
+    max = 999999
   }
 }
 
-variable "droplet_image" {
-  type        = string
-  description = "Which Droplet image would you like to use for your workspace?"
-  default     = "ubuntu-22-04-x64"
-  validation {
-    condition     = contains(["ubuntu-22-04-x64", "ubuntu-20-04-x64", "fedora-36-x64", "fedora-35-x64", "debian-11-x64", "debian-10-x64", "centos-stream-9-x64", "centos-stream-8-x64", "rockylinux-8-x64", "rockylinux-8-4-x64"], var.droplet_image)
-    error_message = "Value must be ubuntu-22-04-x64, ubuntu-20-04-x64, fedora-36-x64, fedora-35-x64, debian-11-x64, debian-10-x64, centos-stream-9-x64, centos-stream-8-x64, rockylinux-8-x64 or rockylinux-8-4-x64."
+data "coder_parameter" "droplet_image" {
+  name    = "Which Droplet image would you like to use for your workspace?"
+  default = "ubuntu-22-04-x64"
+  type    = "string"
+  mutable = false
+  option {
+    name  = "Ubuntu 22.04"
+    value = "ubuntu-22-04-x64"
+  }
+  option {
+    name  = "Ubuntu 20.04"
+    value = "ubuntu-20-04-x64"
+  }
+  option {
+    name  = "Fedora 36"
+    value = "fedora-36-x64"
+  }
+  option {
+    name  = "Fedora 35"
+    value = "fedora-35-x64"
+  }
+  option {
+    name  = "Debian 11"
+    value = "debian-11-x64"
+  }
+  option {
+    name  = "Debian 10"
+    value = "debian-10-x64"
+  }
+  option {
+    name  = "CentOS Stream 9"
+    value = "centos-stream-9-x64"
+  }
+  option {
+    name  = "CentOS Stream 8"
+    value = "centos-stream-8-x64"
+  }
+  option {
+    name  = "Rocky Linux 8"
+    value = "rockylinux-8-x64"
+  }
+  option {
+    name  = "Rocky Linux 8.4"
+    value = "rockylinux-8-4-x64"
   }
 }
 
-variable "droplet_size" {
-  type        = string
-  description = "Which Droplet configuration would you like to use?"
-  default     = "s-1vcpu-1gb"
-  validation {
-    condition     = contains(["s-1vcpu-1gb", "s-1vcpu-2gb", "s-2vcpu-2gb", "s-2vcpu-4gb", "s-4vcpu-8gb", "s-8vcpu-16gb"], var.droplet_size)
-    error_message = "Value must be s-1vcpu-1gb, s-1vcpu-2gb, s-2vcpu-2gb, s-2vcpu-4gb, s-4vcpu-8gb or s-8vcpu-16gb."
+data "coder_parameter" "droplet_size" {
+  name    = "Which Droplet configuration would you like to use?"
+  default = "s-1vcpu-1gb"
+  type    = "string"
+  mutable = false
+  option {
+    name  = "s-1vcpu-1gb"
```

Comment:
We have the ability to use friendlier names now! Thoughts on `1 Core, 1GB Memory`

Reply:
This is a work in progress. I will update to use friendly names. Thanks for reminding me, though.

Reply:
Addressed.